lua/packer/clean.lua

```diff
@@ -67,10 +67,9 @@ local clean_plugins = function(_, plugins, results)
       if await(display.ask_user('Removing the following directories. OK? (y/N)', lines)) then
         results.removals = dirty_plugins
         log.debug('Removed ' .. vim.inspect(dirty_plugins))
-        if util.is_windows then
-          for _, path in ipairs(dirty_plugins) do os.execute('cmd /C rmdir /S /Q ' .. path) end
-        else
-          os.execute('rm -rf ' .. table.concat(dirty_plugins, ' '))
+        for _, path in ipairs(dirty_plugins) do
+          local result = vim.fn.delete(path, 'rf')
```

Comment:
Is `vim.fn.delete` reliable on Windows? We've had trouble with other Vim functions on Windows in the past (e.g. https://github.com/wbthomason/packer.nvim/issues/49) and using `rmdir` was added by a Windows user in https://github.com/wbthomason/packer.nvim/commit/37a19658eeff5710a7ac074deb58f5e67476a675.

I'm fine switching to `vim.fn.delete` if it's known to work, but - as usual with Windows things - I do not personally know this.

Reply:
I had a look at the implementation and it seems to be a somewhat thin wrapper around `libuv` functions. It looks [fairly well tested](https://github.com/neovim/neovim/blob/master/test/functional/legacy/delete_spec.lua), too.

I tried the function on Windows and it worked fine in most cases. The only time things got weird was when I kept a folder open in the explorer: `delete()` got rid of all the files in the test folder but couldn't remove the folder hierarchy. `rmdir` exhibits the same behavior, though, I think that's just a Windows limitation in general.

I'm happy to switch back to `rmdir` if you think it's safer.

Reply:
If `rmdir` has the same unusual failure case, I'm ok with relying on `fn.delete` (given the test coverage, etc.). Thanks for testing that out!